README.md

```diff
@@ -1 +1,142 @@
 # servo-rancher
+Optune servo driver for Rancher (v1.6)
+
+This driver supports an application deployed via Rancher as a Stack.
+
+A Rancher Stack may contain multiple containers, each of which may individually be modified by the adjust driver.
+Each container in the Stack is considered a separately adjustable component.
+
+For each component of the application, the following settings are automatically available when 'adjust --info' is run:
+* scale (number of container instances)
+* memory reservation
+* milli-cpu reservation
+* environment variables
+
+A configuration file must be present to define scale of variable modification config.yaml, any custom component settings in it are also returned by 'adjust --info' and map to environment variables of the matching container.
+
+Currently the following parameters are supported:
+
+environment variables:
+* MEMORY: defines the java heap size in Kilobytes
+
+"Docker Compose" variables:
+* cpu-reservation: millicpu share of the CPU resources
+* memory-reservation: memory reserved reserved in bytes/1024 (scale parameter will adjust)
+
+The following is an example config.yaml document, with examples of the available settings:
+```
+## These elements are required
+# Config group for rancher based adjust driver
+rancher:
+# Rancher Environment(UI) or Project(API) name
+  project: "Default"
+# Target Rancher Stack name
+  stack: "http-test"
+## These elements are optional
+# API key - if not passed as a secret via rancher (api_key)
+  api_key: "ABCDEFG"
+# API secret key - if not passed as a secret via rancher (api_secret)
+  api_secret: "HIJKLMNO"
```

Comment:
Which takes precedence if both a secret and an env var are provided?

Reply:
config file > /var/secrets/api_secrets > env

This is more or less arbitrary based on it was the simplest thing to implement.